### PR TITLE
UPSTREAM: 65215: add counter for pod failures per node

### PR DESCRIPTION
**Pull request by mfojtik**
Picks https://github.com/kubernetes/kubernetes/pull/65215

This will add useful operational metrics for daemonsets fighting with kubelet for pod creation.
It can also be used to alert as when this counter increases it likely mean a conflict in taints/tolerations/ nodeSelector or the node can be under DiskPressure.

/cc @smarterclayton

### CODE REVIEW

File: vendor/k8s.io/kubernetes/pkg/controller/daemon/metrics.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+)
+
+var (
+	podFailedPerNodeNumber = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Subsystem: daemonControllerSubsystem,
+			Name:      podFailedPerNodeNumberKey,
+			Help:      "Number of pods observed as failed for particular daemon set per node.",
+		},
+		[]string{"daemon", "node_name"},
```
Comment (liggitt) [inline vendor/k8s.io/kubernetes/pkg/controller/daemon/metrics.go]: I did not expect a metric with a label per resource... the cardinality is way too high, right?

Comment (liggitt) [reply vendor/k8s.io/kubernetes/pkg/controller/daemon/metrics.go]: basically the same comment as https://github.com/kubernetes/kubernetes/pull/65215#issuecomment-398406184

Comment (mfojtik) [reply vendor/k8s.io/kubernetes/pkg/controller/daemon/metrics.go]: @liggitt @smarterclayton was ok with the cardinality

**Comment by soltysh** [comment]
/hold 
for the rebase to land

**Comment by soltysh** [comment]
@mfojtik mind rebasing this one

**Pull request closed without merging**